estate/models/estate_property.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "This is a estate property app"
+    _order = 'id desc'
+    
+    @api.depends("garden_area","living_area")
+    def _compute_area(self):
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+            
+    @api.depends('offer_ids')
+    def _compute_best_offer(self):
+        for record in self:
+            if(record.offer_ids):
+                record.best_offer = max(record.offer_ids.mapped('price'))
+            else:
+                record.best_offer = 0
+    
+    @api.onchange('is_garden')
+    def _onchange_garden(self):
+        if(self.is_garden is True):
```

Comment:
```suggestion
        if self.is_garden:
```
By default it will only pass if true, if you don't know try by performing it.